### PR TITLE
Adding validations for gadgets configs & expandable advanced settings

### PR DESCRIPTION
**Pull request by senthuran16**
## Purpose
Grouping the advanced settings properties (that are optional to configure) together and making the group expandable

## Goals
Increasing the gadget configuration form's readability.

## Approach
image

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/dashboards-web-component/src/gadgets-generation-wizard/components/ChartConfigurator.jsx [hunk 1]
```diff
@@ -131,7 +139,14 @@ class ChartConfigurator extends Component {
         return (<div />);
     }
 
-    submitGadget() {
+    /**
```
Comment (lasanthaS) [inline components/dashboards-web-component/src/gadgets-generation-wizard/components/ChartConfigurator.jsx]: Remove this comment

File: components/dashboards-web-component/src/gadgets-generation-wizard/components/chartPropertyGenerators/main/LineAreaBarChart.jsx [hunk 2]
```diff
@@ -226,104 +227,116 @@ class LineAreaBar extends Component {
...
-                    <br />
-                    <TextProperty
-                        id="yAxisTickAngle"
-                        value={this.state.configuration.style.yAxisTickAngle}
-                        fieldName="Angle of the y axis ticks"
-                        number
-                        onChange={(id, value) => this.handleMainChartStylePropertyChange(id, value)}
-                        fullWidth
+                <Card
+                    style={(this.state.expandAdvanced) ? ({ padding: 30 }) : ({ padding: 10 })}
```
Comment (lasanthaS) [inline components/dashboards-web-component/src/gadgets-generation-wizard/components/chartPropertyGenerators/main/LineAreaBarChart.jsx]: No need of parenthesis here.

File: components/dashboards-web-component/src/gadgets-generation-wizard/components/chartPropertyGenerators/main/PieChart.jsx [hunk 3]
```diff
@@ -247,40 +176,124 @@ class Pie extends Component {
...
-                    />
-                    <br />
-                    <ColorProperty
-                        id="legendTextColor"
-                        value={this.state.configuration.style.legendTextColor}
-                        fieldName="Text color of the text in the legend"
-                        onChange={(id, value) => this.handleChartStylePropertyChange(id, value)}
-                        fullWidth
+                <Card
+                    style={(this.state.expandAdvanced) ? ({ padding: 30 }) : ({ padding: 10 })}
```
Comment (lasanthaS) [inline components/dashboards-web-component/src/gadgets-generation-wizard/components/chartPropertyGenerators/main/PieChart.jsx]: No need of parenthesis here.

File: components/dashboards-web-component/src/gadgets-generation-wizard/components/ChartConfigurator.jsx (hunk 1 above)
Comment (senthuran16) [reply components/dashboards-web-component/src/gadgets-generation-wizard/components/ChartConfigurator.jsx]: Resolved

**Pull request merged**